### ISSUE TITLE
New Relic Browser support

### ISSUE THREAD

**Issue by adamdicarlo** [newrelic/go-agent#61]
We really want to be able to deploy the Browser agent from Golang backends like one can do with all the other backend languages. Is this being worked on?

We currently have to create Browser apps manually, and ship some janky JS to determine which Browser appId to use, and use that instead of the hardcoded appId that comes in the snippet. So when updating the Browser agent, we not only have to re-copy-paste it, we have to go back and modify part of it again.

To make this work without having several copies of the snippet in our repo, we actually have a separate webpack entry that gets compiled and inlined into a script tag in the `<head>`... it looks like this:

```javascript
// `import` automatically enables strict mode, which breaks the New Relic Browser snippet
const { SERVER_TYPE, SERVER_TYPES } = require('constants/Environment')

const NR_APP_IDS = {
  [SERVER_TYPES.PRODUCTION]: '......36',
  [SERVER_TYPES.LOCAL]: '......18',
  [SERVER_TYPES.TEST]: '......22',
  [SERVER_TYPES.EA]: '......98'
}

const NR_APP_ID = NR_APP_IDS[SERVER_TYPE]

/* eslint-disable */
// New Relic Browser snippet
// <most of Browser agent loader omitted for brevity>
;NREUM.info = {/* ... */ applicationID: NR_APP_ID }
```

That's pretty sad, but it's going to get worse when the app in question needs to have separate deployments for individual enterprise customers -- we'll need to manually create each browser app and add each Browser appId to the list.

I'm curious about the challenges behind making this work. Is there anything we could manually do to achieve an at least half-baked version of first-class Browser support?

Any information or help would be appreciated!

**Comment by LawnGnome** [comment]
We've been working our way through bringing the Go agent to feature parity with the other New Relic language agents, and honestly, the main reason we haven't tackled browser yet is that we wanted to ship other, more frequently requested things like CAT support.

However, I'm pleased to say that we have started working on this internally, and I expect this will be in the next Go agent release in the form of an explicit function on the `Transaction` interface that will return the appropriate blob of JavaScript to make Browser work, which you can then template or interpolate into your output.

So, basically: what you're doing today in terms of copy/pasting the JS from the Browser UI is the right thing to do, but watch this space, because it'll become better soon. 😄

**Comment by willnewrelic** [comment]
Hi @adamdicarlo 

We are excited to say that we have just released version 2.5.0 with browser support!  The Transaction now has a BrowserTimingHeader method to return the JavaScript.

https://github.com/newrelic/go-agent/blob/master/CHANGELOG.md#250
